src/Microsoft.AspNet.Routing/Template/TemplateRoute.cs

```diff
@@ -91,6 +94,11 @@ public class TemplateRoute : INamedRouter
             get { return _dataTokens; }
         }
 
+        public IReadOnlyDictionary<string, bool> Optionals
```

Comment:
This is much too specific to your needs. Just make `_parsedTemplate` into a public property get-only